### PR TITLE
[#81] Implement 'hit diff' command

### LINKED ISSUE #81: Implement `hit diff` command with pretty diff
This command will show pretty diff output like in the image below:

Screenshot from 2019-07-13 22-20-17

The command to do this:

```
git diff HEAD~1 --color=always | diff-highlight | less -rFX
```

However, `diff-highlight` requires installation. Instructions:

1. Ubuntu: https://qiita.com/SakaiYuki/items/2814d417d0bec59046bb
2. Macos: https://www.viget.com/articles/dress-up-your-git-diffs-with-word-level-highlights/

This to look for:

1. Check whether `diff-highlight` is on path and call ordinary diff

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #81

I have `diff-highlight` installed locally and it works for me!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ Here is an example of how you can see the list of issues and the issue descripti
 To start using `hit` make sure that you have the following tools installed on your machine:
 
 * [ `git`][git] — `hit` is a wrapper around `git`
+* `diff-highlight` — for pretty output of the `hit diff` command
```
Comment (vrom911) [inline README.md]: Should somehow highlight that this is optional, but recommended.

File: src/Hit/Git.hs [hunk 2]
```diff
@@ -191,3 +196,23 @@ issueFromBranch =
...
+    -- Add all untracked file to index so they will appear in diff
+    addUntrackedFiles :: IO [Text]
+    addUntrackedFiles = do
+        untrackedFiles <- lines <$> "git" $| ["ls-files", "--others", "--exclude-standard"]
+        for_ untrackedFiles $ \file -> void $ "git" $| ["add", file]
+        pure untrackedFiles
+
+    -- Return untracked files back to not spoil git state and have unexpected behavior
+    removeUntrackedFiles :: [Text] -> IO ()
+    removeUntrackedFiles = mapM_ $ \file -> void $ "git" $| ["reset", file]
```
Comment (vrom911) [inline src/Hit/Git.hs]: This won't affect files that were tracked before, will it?

File: README.md (hunk 1 above)
Comment (chshersh) [reply README.md]: Good idea! I can add a bold prefix like `**Optional**`. What do you think?

File: src/Hit/Git.hs (hunk 2 above)
Comment (chshersh) [reply src/Hit/Git.hs]: @vrom911 Not, it doesn't affect files that are already tracked. List of `[Text]` here is the files that weren't tracked and returned by the `addUntrackedFiles` function. This function works only with untracked files.

**Pull request merged**